### ISSUE TITLE
0.9.4-rc.1: kube-aws up --export does not export userdata

### ISSUE THREAD

**Issue by cheungpat** [coreos/kube-aws#300]
The `kube-aws up --export` allows me to export the stack template without creating the stack in CloudFormation. However, it does not generate the instance userdata for controller and workers.

```shell
$ kube-aws up --export --s3-uri s3://bucket
Exporting name.stack-template.json
```

After this command is run, the stack template is saved in the local file system but `userdata-controller` and `userdata-worker` are not. This is because `--export` causes the `up` command to only save the stack templates but not userdata.

One quick solution is to save userdata in a file named like `{stackname}.userdata-{controller,worker}` (which is similar to the format: `{stackname}.stack-template.json`)

**Comment by mumoshu** [comment]
@cheungpat Good catch!

I'd rather like to put all the necessary assets into the new directory named `export/` so that its content would look like:

- `export/`
  - `userdata/`
     - `cloud-config-controller`
     - `cloud-config-worker`
  - `stack.json` (former `{stackame}.stack-template.json` renamed to clarify that it is different from the `stackc-template.json` which is a golang text-template to render json. I have been desired to rename it like this 😃 )

**Comment by cheungpat** [comment]
I would like to work on this with @mumoshu’s suggestion on file naming. Should I make a pull request or should I wait longer for community feedback?

**Comment by mumoshu** [comment]
@cheungpat Thanks for checking in.
The former would be nice because then we can make it just work at first and then gradually improve it if necessary.
I'm looking forward to your pull request :+1:

**Comment by mumoshu** [comment]
Hey @cheungpat, I've addressed this in #315 because it is required in process of E2E testing 👍 
Please see the description of the PR for more information and especially about which/to where files are exported.
I hope you like it!

**Comment by cheungpat** [comment]
Oh, you have beat me to it. Thanks for your help! I will take a look.
